tfhe-zk-pok/src/proofs/pke.rs

```diff
@@ -240,9 +240,18 @@ pub struct Proof<G: Curve> {
     c_hat: G::G2,
     c_y: G::G1,
     pi: G::G1,
-    c_hat_t: Option<G::G2>,
-    c_h: Option<G::G1>,
-    pi_kzg: Option<G::G1>,
+    optional_fields: Option<ProofOptionalFields<G>>,
+}
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+#[serde(bound(
+    deserialize = "G: Curve, G::G1: serde::Deserialize<'de>, G::G2: serde::Deserialize<'de>",
+    serialize = "G: Curve, G::G1: serde::Serialize, G::G2: serde::Serialize"
+))]
+struct ProofOptionalFields<G: Curve> {
```

Comment:
ComputeLoadProofFields, field name compute_load_proof_fields, maybe with a comment to indicate that those are only populated in the case of a proof with ComputeLoad::Proof

would better reflect what this is, I would wait for the https://github.com/zama-ai/tfhe-rs/pull/1597 PR to be merged first

this will need some upgrade code to make compatbile